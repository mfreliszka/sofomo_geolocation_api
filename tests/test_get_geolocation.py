"""Module containing tests for the get_geolocation endpoint."""

from fastapi import FastAPI, status
import pytest

from syrupy.assertion import SnapshotAssertion


@pytest.mark.anyio
async def test_get_existing_geolocation(
    app: FastAPI,
    IPGeolocation1_InDB_Model: pytest.fixture,
    snapshot: SnapshotAssertion,
    valid_ip1: pytest.fixture,
    httpx_async_client: pytest.fixture,
):
    """Test retrieving an existing geolocation from the database."""
    response = await httpx_async_client.get(
        app.url_path_for("get_geolocation_from_database"),
        params={"ip_address": valid_ip1},
    )

    assert response.json() == snapshot, "Actual response did not match snapshot"
    assert response.status_code == status.HTTP_200_OK, (
        f"Response status code: {response.status_code}"
    )
    assert response.headers["Content-Type"] == "application/json", (
        f"Response content type: {response.headers['Content-Type']}"
    )


@pytest.mark.anyio
async def test_get_not_existing_geolocation(
    app: FastAPI,
    snapshot: SnapshotAssertion,
    non_existent_ip: pytest.fixture,
    httpx_async_client: pytest.fixture,
):
    """Test retrieving not existing geolocation from the database."""
    response = await httpx_async_client.get(
        app.url_path_for("get_geolocation_from_database"),
        params={"ip_address": non_existent_ip},
    )

    assert response.json() == snapshot, "Actual response did not match snapshot"
    assert response.status_code == status.HTTP_404_NOT_FOUND, (
        f"Response status code: {response.status_code}"
    )
    assert response.headers["Content-Type"] == "application/json", (
        f"Response content type: {response.headers['Content-Type']}"
    )


@pytest.mark.anyio
async def test_get_invalid_ip_address(
    app: FastAPI,
    snapshot: SnapshotAssertion,
    invalid_ip: pytest.fixture,
    httpx_async_client: pytest.fixture,
):
    """Test retrieving invalid IP address from the database."""
    response = await httpx_async_client.get(
        app.url_path_for("get_geolocation_from_database"),
        params={"ip_address": invalid_ip},
    )

    assert response.json() == snapshot, "Actual response did not match snapshot"
    assert response.status_code == status.HTTP_400_BAD_REQUEST, (
        f"Response status code: {response.status_code}"
    )
    assert response.headers["Content-Type"] == "application/json", (
        f"Response content type: {response.headers['Content-Type']}"
    )
