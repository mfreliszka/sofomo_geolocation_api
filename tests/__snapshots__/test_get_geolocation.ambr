# serializer version: 1
# name: test_get_existing_geolocation[asyncio]
  dict({
    'city': 'Mountain View',
    'continent_code': 'NA',
    'continent_name': 'North America',
    'country_code': 'US',
    'country_name': 'United States',
    'id': 1,
    'ip': '8.8.8.8',
    'latitude': 37.422,
    'longitude': -122.084,
    'region_code': 'CA',
    'region_name': 'California',
    'type': 'IPv4',
    'updated_at': '2020-01-01T00:00:00',
    'zip': '94043',
  })
# ---
# name: test_get_existing_geolocation[trio]
  dict({
    'city': 'Mountain View',
    'continent_code': 'NA',
    'continent_name': 'North America',
    'country_code': 'US',
    'country_name': 'United States',
    'id': 1,
    'ip': '8.8.8.8',
    'latitude': 37.422,
    'longitude': -122.084,
    'region_code': 'CA',
    'region_name': 'California',
    'type': 'IPv4',
    'updated_at': '2020-01-01T00:00:00',
    'zip': '94043',
  })
# ---
# name: test_get_invalid_ip_address[asyncio]
  dict({
    'detail': "Invalid IP address: 'invalid_ip'",
  })
# ---
# name: test_get_invalid_ip_address[trio]
  dict({
    'detail': "Invalid IP address: 'invalid_ip'",
  })
# ---
# name: test_get_not_existing_geolocation[asyncio]
  dict({
    'detail': "Geolocation for IP '1.1.1.1' not found in database.",
  })
# ---
# name: test_get_not_existing_geolocation[trio]
  dict({
    'detail': "Geolocation for IP '1.1.1.1' not found in database.",
  })
# ---
